openfisca_core/tools/test_runner.py

```diff
@@ -0,0 +1,232 @@
+        (copy.deepcopy(key), value)
+        for key, value in data.iteritems()
+        ))
+    yaml.add_representer(folded_unicode, lambda dumper, data: dumper.represent_scalar(u'tag:yaml.org,2002:str',
+        data, style='>'))
+    yaml.add_representer(literal_unicode, lambda dumper, data: dumper.represent_scalar(u'tag:yaml.org,2002:str',
+        data, style='|'))
+    yaml.add_representer(np.ndarray, lambda dumper, data: dumper.represent_list(data.tolist()))
+    yaml.add_representer(periods.Instant, lambda dumper, data: dumper.represent_scalar(u'tag:yaml.org,2002:str', str(data)))
+    yaml.add_representer(periods.Period, lambda dumper, data: dumper.represent_scalar(u'tag:yaml.org,2002:str', str(data)))
+    yaml.add_representer(tuple, lambda dumper, data: dumper.represent_list(data))
+    yaml.add_representer(unicode, lambda dumper, data: dumper.represent_scalar(u'tag:yaml.org,2002:str', data))
+
+    return yaml
+
+
+_config_yaml(yaml)
+
+
+# Exposed methods
+
+def generate_tests(tax_benefit_system, path, options = {}):
+    """
+    Generates a lazy iterator of all the yaml tests contained in a file or a directory.
+
+    :parameters: Same as :meth:`run_tests`
+
+    :return: a generator of yaml tests
+
+    """
+
+    if os.path.isdir(path):
+        return _generate_tests_from_directory(tax_benefit_system, path, options)
+    else:
+        return _generate_tests_from_file(tax_benefit_system, path, options)
+
+
+def run_tests(tax_benefit_system, path, options = {}):
+    """
+    Runs all the yaml tests contained in a file or a directory.
+
+    If `path` is a directory, subdirectories will be recursively explored.
+
+    :param TaxBenefitSystem tax_benefit_system: the tax-benefit system to use to run the tests
+    :param str path: the path towards the file or directory containing thes tests. If it is a directory, subdirectories will be recursively explored.
+    :param dict options: a dictionnary of options. See more details below.
+
+    :raises AssertionError: if a test does not pass
+
+    :return: the number of sucessful tests excecuted
+
+    **Testing options**:
+
+    =============================  ==========  ======================================================================
+    Key                            Type        Role
+    =============================  ==========  ======================================================================
+    force                          ``bool``    force testing of tests with "ignore" flag
+    verbose                        ``bool``    increase output verbosity
+    name_filter                    ``str``     only run tests with name_filter in their name, file name, or keywords
+    default_absolute_error_margin  ``float``   set the absolute error margin to use for test which don't redefine one
+    default_relative_error_margin  ``float``   set the relative error margin to use for test which don't redefine one
```

Comment:
If “the same options” as in “is just pass-through”, then let the doc be a pass-through too indeed. I doubt there will be strong usage of the module apart from the CLI anyway, so let's prevent outdated doc.